docs/src/literate-tutorials/linear_elasticity.jl

```diff
@@ -1,3 +1,354 @@
+#
+# Here we use linear triangular elements (also called constant strain triangles).
+# The vector valued shape functions are constructed by raising the interpolation
+# to the power `dim` (the dimension) since the displacement field has one component in each
+# spatial dimension.
+dim = 2
+order = 1 # linear interpolation
+ip = Lagrange{RefTriangle, order}()^dim; # vector valued interpolation
+
+# In order to evaluate the integrals, we need to specify the quadrature rules to use. Due to the
+# linear interpolation, a single quadrature point suffices, both inside the cell and on the facet.
+# In 2d, a facet is the edge of the element.
+qr = QuadratureRule{RefTriangle}(1) # 1 quadrature point
+qr_face = FacetQuadratureRule{RefTriangle}(1);
+
+# Finally, we collect the interpolations and quadrature rules into the `CellValues` and `FacetValues`
+# buffers, which we will later use to evaluate the integrals over the cells and facets.
+cellvalues = CellValues(qr, ip)
+facetvalues = FacetValues(qr_face, ip);
+
+# ### Degrees of freedom
+# For distributing degrees of freedom, we define a `DofHandler`. The `DofHandler` knows that
+# `u` has two degrees of freedom per node because we vectorized the interpolation above.
+dh = DofHandler(grid)
+add!(dh, :u, ip)
+close!(dh);
+
+# ### Boundary conditions
+# We set Dirichlet boundary conditions by fixing the motion normal to the bottom and left
+# boundaries. The last argument to `Dirichlet` determines which components of the field should be
+# constrained. If no argument is given, all components are constrained by default.
+ch = ConstraintHandler(dh)
+add!(ch, Dirichlet(:u, getfacetset(grid, "bottom"), (x, t) -> 0.0, 2))
+add!(ch, Dirichlet(:u, getfacetset(grid, "left"),   (x, t) -> 0.0, 1))
+close!(ch);
+
+# In addition, we will use Neumann boundary conditions on the top surface, where
+# we add a traction vector of the form
+# ```math
+# \boldsymbol{t}_\mathrm{N}(\boldsymbol{x}) = (20e3) x_1 \boldsymbol{e}_2
+# ```
+traction(x) = Vec(0.0, 20e3 * x[1]);
+
+# On the right boundary, we don't do anything, resulting in a zero traction Neumann boundary.
+# In order to assemble the external forces, $f_i^\mathrm{ext}$, we need to iterate over all
+# facets in the relevant facetset. We do this by using the [`FacetIterator`](@ref).
+
+function assemble_external_forces!(f_ext, dh, facetset, facetvalues, prescribed_traction)
+    ## Create a temporary array for the facet's local contributions to the external force vector
+    fe_ext = zeros(getnbasefunctions(facetvalues))
+    for face in FacetIterator(dh, facetset)
+        ## Update the facetvalues to the correct facet number
+        reinit!(facetvalues, face)
+        ## Reset the temporary array for the next facet
+        fill!(fe_ext, 0.0)
+        for qp in 1:getnquadpoints(facetvalues)
+            ## Calculate the global coordinate of the quadrature point.
+            x = spatial_coordinate(facetvalues, qp, getcoordinates(face))
```

Comment:
Does it make sense to lift out the `getcoordinates`  call outside the `qp` loop? Not really for performance but to show that this is independent of the qp.